test/server/http/admin_test.cc

```diff
@@ -68,13 +84,27 @@ TEST_F(AdminFilterTest, AdminProfiler) {
 
 #endif
 
-TEST_F(AdminFilterTest, AdminBadProfiler) {
+TEST_F(AdminInstanceTest, AdminBadProfiler) {
   Buffer::OwnedImpl data;
-  AdminImpl admin_bad_profile_path("/dev/null",
-                                   TestEnvironment::temporaryPath("some/unlikely/bad/path.prof"),
-                                   Network::Utility::resolveUrl("tcp://127.0.0.1:0"), server_);
+  AdminImpl admin_bad_profile_path(
+      "/dev/null", TestEnvironment::temporaryPath("some/unlikely/bad/path.prof"), "",
+      Network::Test::getSomeLoopbackAddress(Network::Address::IpVersion::v4), server_);
   admin_bad_profile_path.runCallback("/cpuprofiler?enable=y", data);
   EXPECT_FALSE(Profiler::Cpu::profilerEnabled());
 }
 
+TEST_F(AdminInstanceTest, WriteAddressToFile) {
+  std::ifstream address_file(address_out_path_);
+  std::string address_from_file;
+  std::getline(address_file, address_from_file);
+  EXPECT_STREQ(admin_.socket().localAddress()->asString().c_str(), address_from_file.c_str());
+}
+
+TEST_F(AdminInstanceTest, AdminBadAddressOutPath) {
+  std::string bad_path = TestEnvironment::temporaryPath("some/unlikely/bad/path/admin.address");
+  AdminImpl admin_bad_address_out_path(
+      "/dev/null", cpu_profile_path_, bad_path,
+      Network::Test::getSomeLoopbackAddress(Network::Address::IpVersion::v4), server_);
+  EXPECT_FALSE(std::ifstream(bad_path));
```

Comment:
The logger could be mocked and the log statement EXPECTed, but, the way the logger object is brought into existence in many places including in AdminImpl makes this difficult.  It's super convenient to be able to inherit in any class from Logger::Loggable<LogIdType> and just get the magic log() method but this kills the flexibility of being able to pass in a mocked logger.  Some test code added to the logger could provide a special mode where you can check what's been logged, but probably this is nontrivial and best done as a separate PR.